src/components/ChangePasswordForm.js

```diff
@@ -0,0 +1,186 @@
+        <div className='sp-change-password-form'>
+          <div className="row" >
+            <div className="col-sm-offset-4 col-xs-12 col-sm-4" spIf="form.sent">
+              <p className="alert alert-warning text-center" spIf="form.processing">
+                We are verifying your change password request...
+              </p>
+              <p className="alert alert-success" spIf="form.success">Your new password has been set. Please <LoginLink />.</p>
+              <div className="alert alert-danger" spIf="form.error">
+                This password reset link is not valid.  You may request another link by <a href="/forgot">clicking here</a>.
+              </div>
+            </div>
+            <div className="col-xs-12" spIf="!form.sent">
+              <div className="form-horizontal">
+                <div className="form-group">
+                  <label htmlFor="spPassword" className="col-xs-12 col-sm-4 control-label">New Password</label>
+                  <div className="col-xs-12 col-sm-4">
+                    <input id="spPassword" type="password" name="password" className="form-control" placeholder="New Password" required />
+                  </div>
+                </div>
+                <div className="form-group">
+                  <label htmlFor="spConfirmPassword" className="col-xs-12 col-sm-4 control-label">Confirm New Password</label>
+                  <div className="col-xs-12 col-sm-4">
+                    <input id="spConfirmPassword" type="password" name="confirmPassword" className="form-control" placeholder="Confirm New Password" required />
+                  </div>
+                </div>
+                <div className="form-group">
+                  <div className="col-sm-offset-4 col-sm-4">
+                    <p className="alert alert-danger" spIf="form.error"><span spBind="form.errorMessage" /></p>
+                    <button type="submit" className="btn btn-primary">Set New Password</button>
+                  </div>
+                </div>
+              </div>
+            </div>
+          </div>
+        </div>
+      </ChangePasswordForm>
+    );
+  }
+}
+
+@ReactMixin.decorate(History)
+export default class ChangePasswordForm extends React.Component {
+  state = {
+    spToken: null,
+    fields: {
+      password: '',
+      confirmPassword: ''
+    },
+    errorMessage: null,
+    isFormProcessing: false,
+    isFormSent: false
+  };
+
+  constructor(...args) {
+    super(...args);
+    this.state.spToken = this.props.spToken;
+  }
+
+  onFormSubmit(e) {
+    e.preventDefault();
+
+    var next = (err, data) => {
+      if (err) {
+        return this.setState({
+          isFormProcessing: false,
+          errorMessage: err.message
+        });
+      }
+
+      // If the user didn't specify any data,
+      // then simply default to what we have in state.
+      data = data || this.state.fields;
+
+      if (data.confirmPassword && data.password !== data.confirmPassword) {
```

Comment:
If this is a custom form without the `required` attribute, then it would be possible to leave `confirmPassword` empty and it would not enter this `if` statement.